silo-import/src/silo_import/decompressor.py

```diff
@@ -19,33 +21,99 @@ class NdjsonAnalysis:
 
     record_count: int
     pipeline_versions: set[int]
+    transformed_path: Path
+
+
+def transform_record(record: dict) -> dict:
+    """
+    Transform a single JSON record.
+
+    Args:
+        record: Input record with metadata, sequences, and insertions
+
+    Returns:
+        Transformed record with flattened metadata and combined sequences
+    """
+    result = {}
+
+    metadata = record.get("metadata", {})
+    if isinstance(metadata, dict):
+        result.update(metadata)
+
+    nucleotide_insertions = record.get("nucleotideInsertions", {})
+    amino_acid_insertions = record.get("aminoAcidInsertions", {})
+
+    # Process aligned nucleotide sequences
+    aligned_nuc_seqs = record.get("alignedNucleotideSequences", {})
+    if isinstance(aligned_nuc_seqs, dict):
```

Comment:
These if isinstance are useless - we know it's a dict - otherwise we're screwed no?

Reply:
True. If it's not set at all, the default in the `get` call makes sure that it works.